lib/events/events.go

```diff
@@ -144,16 +148,23 @@ func NewLogger() *Logger {
 
 func (l *Logger) Log(t EventType, data interface{}) {
 	l.mutex.Lock()
-	dl.Debugln("log", l.nextID, t, data)
-	l.nextID++
-	e := Event{
-		ID:   l.nextID,
-		Time: time.Now(),
-		Type: t,
-		Data: data,
-	}
+	dl.Debugln("log", l.nextGlobalID, t, data)
+	l.nextGlobalID++
+
+	now := time.Now()
+
 	for _, s := range l.subs {
 		if s.mask&t != 0 {
+			s.nextSubscriptionID++
+
+			e := Event{
```

Comment:
I'm not sure whether doing this, or creating one `Event` and mutating the `SubscriptionID` field for each subscription, is better. This is neater (IMO), but I'm not sure if it incurs some additional cost.